From fedefea5b2a0b4bda81228beaf3167622c4e0a14 Mon Sep 17 00:00:00 2001
From: Michal Simek <monstr@monstr.eu>
Date: Wed, 3 Oct 2012 13:35:43 +0200
Subject: [PATCH 479/628] usb: gadget: Use module_platform_driver

git://github.com/Xilinx/linux-xlnx.git xilinx-v14.7
commit e115fb8eba32fcd139fb3612812d7a58d138f695

Use shortcuts as much as possible.

Signed-off-by: Michal Simek <monstr@monstr.eu>
Signed-off-by: Liming Wang <liming.wang@windriver.com>
---
 drivers/usb/gadget/xilinx_usbps_udc.c |   17 ++---------------
 1 files changed, 2 insertions(+), 15 deletions(-)

diff --git a/drivers/usb/gadget/xilinx_usbps_udc.c b/drivers/usb/gadget/xilinx_usbps_udc.c
index cf3c32b..ae1c52e 100755
--- a/drivers/usb/gadget/xilinx_usbps_udc.c
+++ b/drivers/usb/gadget/xilinx_usbps_udc.c
@@ -2586,6 +2586,7 @@ static int xusbps_udc_resume(struct platform_device *pdev)
 --------------------------------------------------------------------------*/
 
 static struct platform_driver udc_driver = {
+	.probe   = xusbps_udc_probe,
 	.remove  = __exit_p(xusbps_udc_remove),
 	/* these suspend and resume are not usb suspend and resume */
 	.suspend = xusbps_udc_suspend,
@@ -2596,21 +2597,7 @@ static struct platform_driver udc_driver = {
 	},
 };
 
-static int __init udc_init(void)
-{
-	printk(KERN_INFO "%s (%s)\n", driver_desc, DRIVER_VERSION);
-	return platform_driver_probe(&udc_driver, xusbps_udc_probe);
-}
-
-module_init(udc_init);
-
-static void __exit udc_exit(void)
-{
-	platform_driver_unregister(&udc_driver);
-	printk(KERN_WARNING "%s unregistered\n", driver_desc);
-}
-
-module_exit(udc_exit);
+module_platform_driver(udc_driver);
 
 MODULE_DESCRIPTION(DRIVER_DESC);
 MODULE_AUTHOR(DRIVER_AUTHOR);
-- 
1.7.5.4

